# E91 Quantum Key Distribution

E91 is a quantum key distribution (QKD) protocol that uses quantum entanglement itself as the foundation of security. The security comes directly from quantum correlations between particles.

A source creates entangled Bell pairs and distributes them between Alice & Bob. 

Each person measures their particle using randomly chosen measurement settings. 

Because the particles are entangled, the measurement outcomes become strongly correlated even though the measurements happen separately.

# Why Entanglement Matters Here

Entangled systems contain correlations that cannot be reproduced classically.

This means that if somebody tries to intercept or disturb the particles, the quantum correlations themselves become damaged.

# How Physics Detects Eavesdropping

Suppose Alice and Bob repeatedly measure entangled particles.

Without interference, their measurement outcomes show very strong correlations.

For example, they may repeatedly observe matching outcomes like:

0 → 0
1 → 1

much more strongly than expected classically.

Initially this felt similar to ordinary correlation, like two envelopes secretly containing matching cards.

But the important part is that quantum entanglement produces correlations stronger than classical hidden instructions can explain.

If an eavesdropper tries to:
- measure particles
- intercept communication
- copy quantum states

they must interact with the quantum system itself.

That interaction disturbs the entanglement.

As a result, the correlation patterns between Alice and Bob start weakening.

Instead of observing strong structured correlations consistently, the measurement outcomes become noisier and less correlated.

This change in correlation statistics reveals that the quantum system was disturbed.

# Bell Inequalities

Bell inequalities are mathematical limits describing how strongly classical systems can correlate with each other.

Classical systems behave as if particles carry hidden predetermined instructions because of which classical correlations cannot exceed certain statistical limits.

Entangled quantum systems are different because they can violate Bell inequalities.

This means the correlations between entangled particles become stronger than what classical hidden-variable explanations allow.

E91 checks whether these Bell inequality violations still exist.

If Alice and Bob continue observing strong Bell violations, then the entanglement is still behaving quantum mechanically.

If the Bell violation weakens significantly, it suggests that the system was disturbed, possibly by eavesdropping.

# Comparison With BB84

BB84 mainly relied on random basis choices and measurement disturbance.

E91 is different because the security comes from preserving entanglement and quantum correlations themselves.

In [1]:
from qiskit import QuantumCircuit
from qiskit_aer import AerSimulator

# create circuit
qc = QuantumCircuit(2,2)

# create Bell pair
qc.h(0)
qc.cx(0,1)

# measurements
qc.measure(0,0)
qc.measure(1,1)

# draw circuit
print(qc.draw())

# run simulator
simulator = AerSimulator()

job = simulator.run(qc, shots=1000)

result = job.result()
counts = result.get_counts()

print(counts)

     ┌───┐     ┌─┐   
q_0: ┤ H ├──■──┤M├───
     └───┘┌─┴─┐└╥┘┌─┐
q_1: ─────┤ X ├─╫─┤M├
          └───┘ ║ └╥┘
c: 2/═══════════╩══╩═
                0  1 
{'11': 471, '00': 529}


# Observation

The circuit first created the Bell state:

(|00> + |11>) /sqrt(2)

This meant the two qubits became entangled.

After measurement, the output became approximately:

{'00': 529, '11': 471} (first run)

The important realization was that Alice and Bob mostly obtained matching outcomes.

If Alice measured:
0 → Bob also measured 0
1 → Bob also measured 1

At the same time, the individual outcomes themselves still remained random.

We cannot predict the exact result beforehand but the outcome is still strongly correlated after measurement.

In [2]:
from qiskit import QuantumCircuit
from qiskit_aer import AerSimulator

# create circuit
qc = QuantumCircuit(2,2)

# create Bell pair
qc.h(0)
qc.cx(0,1)

# simulate disturbance / eavesdropping
qc.x(1)

# measurements
qc.measure(0,0)
qc.measure(1,1)

# draw circuit
print(qc.draw())

# run simulator
simulator = AerSimulator()

job = simulator.run(qc, shots=1000)

result = job.result()
counts = result.get_counts()

print(counts)

     ┌───┐          ┌─┐   
q_0: ┤ H ├──■───────┤M├───
     └───┘┌─┴─┐┌───┐└╥┘┌─┐
q_1: ─────┤ X ├┤ X ├─╫─┤M├
          └───┘└───┘ ║ └╥┘
c: 2/════════════════╩══╩═
                     0  1 
{'01': 508, '10': 492}


# Observation After Disturbance

The second circuit introduced disturbance using X gate on Bob’s qubit.

X flipped Bob’s measurement outcomes.

Because of this, the outputs became approximately:

{'01': 508, '10': 492} (first run)

Now Alice and Bob mostly obtained opposite outcomes instead of matching ones.

before disturbance → strong matching correlations
After disturbance → strong opposite correlations